src/grpc_conversions/extensions.rs

```diff
@@ -317,3 +319,36 @@ impl Hash for RetrievedPoint {
         self.id.hash(state)
     }
 }
+
+impl HardwareUsage {
+    pub(crate) fn aggregate_opts(this: &Option<Self>, other: &Option<Self>) -> Option<Self> {
```

Comment:
Shall we make this owned, because they're `Copy` anyway?

```suggestion
    pub(crate) fn aggregate_opts(this: Option<Self>, other: Option<Self>) -> Option<Self> {
```